src/main/scala/wasm/MiniWasmFX.scala

```diff
@@ -0,0 +1,293 @@
+        val funcTy = getFuncType(ty)
+        val I32V(cond) :: newStack = stack
+        val inner = if (cond != 0) thn else els
+        val (inputs, restStack) = newStack.splitAt(funcTy.inps.size)
+        val restK: Cont[Ans] = (retStack, ehs, mkont) =>
+          eval(rest, retStack.take(funcTy.out.size) ++ restStack, frame, kont, mkont, trail, h, ehs)
+        eval(inner, inputs, frame, restK, mkont, restK :: trail, h, ehs)
+      case Br(label) =>
+        trail(label)(stack, ehs, mkont)
+      case BrIf(label) =>
+        val I32V(cond) :: newStack = stack
+        if (cond != 0) trail(label)(newStack, ehs, mkont)
+        else eval(rest, newStack, frame, kont, mkont, trail, h, ehs)
+      case BrTable(labels, default) =>
+        val I32V(cond) :: newStack = stack
+        val goto = if (cond < labels.length) labels(cond) else default
+        trail(goto)(newStack, ehs, mkont)
+      case Return        => trail.last(stack, ehs, mkont)
+      case Call(f)       => evalCall(f, rest, stack, frame, kont, mkont, trail, h, ehs, false)
+      case ReturnCall(f) => evalCall(f, rest, stack, frame, kont, mkont, trail, h, ehs, true)
+      // XXX (GW): consider implementing call_ref too
+      case RefFunc(f) =>
+        // TODO: RefFuncV stores an applicable function, instead of a syntactic structure
+        eval(rest, RefFuncV(f) :: stack, frame, kont, mkont, trail, h, ehs)
+      case ContNew(ty) =>
+        val RefFuncV(f) :: newStack = stack
+        // create a continuation which only performs a function call
+        val idK: Cont[Ans] = (s, ehs, m) => m(s)
+        val k: Cont[Ans] = (s, ehs, mk) => evalCall(f, List(), s, frame, idK, mk, trail, h, ehs, false)
+        eval(rest, TCContV(k) :: newStack, frame, kont, mkont, trail, h, ehs)
+      case Suspend(tag_id) => {
+        val FuncType(_, inps, out) = module.tags(tag_id)
+        val (input, restStack) = stack.splitAt(inps.size)
+        // todo: ensure the specified effect handlers here is correct
+        val kr: Cont[Ans] = (s, ehs, mk) => eval(rest, s ++ restStack, frame, kont, mk, trail, h, ehs)
+        ehs.find(_._1 == tag_id) match {
+          case Some((_, handler)) => handler(TCContV(kr) :: input)
+          case None               => 
+            throw Trap()
+        }
+      }
+      case Resume(kty_id, handlers) => {
+        val (resume: TCContV[Ans]) :: newStack = stack
+        val contTy = module.types(kty_id)
+        val ContType(funcTypeId) = contTy
+        val FuncType(_, inps, out) = module.types(funcTypeId)
+        val (inputs, restStack) = newStack.splitAt(inps.size)
+        val m: MCont[Ans] = (s) => eval(rest, s ++ restStack, frame, kont, mkont, trail, h, ehs)
+        val newEhs: List[(Int, Handler[Ans])] = handlers.map {
+          case Handler(tag_id, handler) => (tag_id, (s) => trail(handler)(s, ehs, mkont))
+        }
+        // resume with new effect handlers installed each time
+        resume.k(inputs, newEhs, m)
```

Comment:
A todo: `newEhs` should be appended to the current `ehs`

Reply:
I somehow think this is okay? Isn't the handlers kept in the meta continuation?

Reply:
Oh nvm I get what you are talking about. The `newEhs` should be "merged" with h since h might contain a tag id not in `newEhs`

Reply:
The comment I left is about 

> A todo: `newEhs` should be appended to the current `ehs`, to make sure effect handlers are kept in nested resume case.

This comment is a treatment about nested resume. The original treatment in this PR drops the current effect handler list(`ehs` parameter) when resume, when resuming, which I believe should be incorrect?